src/bf_dpu_update.py

```diff
@@ -990,12 +993,70 @@ def factory_reset_bios(self):
         self._wait_for_system_power_on(50, 100)
         print()
 
+    def update_oem_fru(self):
+        """
+        Update the OEM FRU data with the provided key-value pairs in the format 'Section:Key=Value'
+        """
+
+        # Define the required fields for the OEM FRU data
+        required_fields = {
+            'Product:Manufacturer',
+            'Product:SerialNumber',
+            'Product:PartNumber',
+            'Product:Version',
+            'Product:Extra',
+            'Product:ManufactureDate',
+            'Product:AssetTag',
+            'Product:GUID'
+        }
+
+        oem_fru_dict = {}
+        provided_fields = set()
+        if self.debug:
+            print("OEM FRU data to be updated:", self.oem_fru)
+        # Process each item in the provided OEM FRU data
+        for item in self.oem_fru:
+            try:
+                section_key, value = item.split('=')
+                section, key = section_key.split(':')
+                combined_key = section + key
+                oem_fru_dict[combined_key] = value
+                provided_fields.add(section_key)
+                if self.debug:
+                    print(f"Updated FRU field: {section_key} with value: {value}")
+            except ValueError:
+                raise ValueError(f"Invalid format for OEM FRU data: {item}. Expected format 'Section:Key=Value'.")
```

Comment:
We'd can add new error num in error_num.py.   Then it will be more clear about all errors we may have in this script.

Reply:
Done